core/os/os2/process_linux.odin

```diff
@@ -1,68 +1,268 @@
+
+		if .PPid in selection {
+			ppid_str := stats[strings.index_byte(stats, ' '):]
+			if ppid, ok := strconv.parse_int(ppid_str); ok {
+				info.ppid = ppid
+			}
+		}
+
+		if .Priority in selection {
+			// On 4th field. Priority is field 18 and niceness is field 19.
+			for i := 4; i < 19; i += 1 {
+				stats = stats[strings.index_byte(stats, ' ') + 1:]
+			}
+			nice_str := stats[strings.index_byte(stats, ' '):]
+			if nice, ok := strconv.parse_int(nice_str); ok {
+				info.priority = nice
+			}
+		}
+	}
+
+	cmdline_if: if selection & {.Command_Line, .Command_Args, .Executable_Path} != {} {
+		_ = fmt.bprintf(path_slice, "/proc/%d/cmdline", pid)
+		cmdline_bytes := _read_entire_pseudo_file(path_cstr, temp_allocator()) or_return
+		if len(cmdline_bytes) == 0 {
+			break cmdline_if
+		}
+		cmdline := string(cmdline_bytes)
+
+		terminator := strings.index_byte(cmdline, 0)
+		if .Executable_Path in selection {
+			info.executable_path = strings.clone(cmdline[:terminator], allocator)
+		}
+		if .Command_Line in selection {
+			info.command_line = strings.clone(cmdline[:terminator], allocator)
+		}
+
+		if .Command_Args in selection {
+			// skip to first arg
+			cmdline = cmdline[terminator + 1:]
+
+			arg_list := make([dynamic]string, allocator)
+			for len(cmdline) > 0 {
+				terminator = strings.index_byte(cmdline, 0)
+				append(&arg_list, strings.clone(cmdline[:terminator], allocator))
+				cmdline = cmdline[terminator + 1:]
+			}
+			info.command_args = arg_list[:]
+		}
+	}
+
+	if .Environment in selection {
+		_ = fmt.bprintf(path_slice, "/proc/%d/environ", pid)
+		if env_bytes, env_err := _read_entire_pseudo_file(path_cstr, temp_allocator()); env_err == nil {
+			env := string(env_bytes)
+
+			env_list := make([dynamic]string, allocator)
+			for len(env) > 0 {
+				terminator := strings.index_byte(env, 0)
+				if terminator == -1 || terminator == 0 {
+					break
+				}
+				append(&env_list, strings.clone(env[:terminator], allocator))
+				env = env[terminator + 1:]
+			}
+			info.environment = env_list[:]
+		}
+	}
+
 	return
 }
 
 @(private="package")
 _process_info_by_handle :: proc(process: Process, selection: Process_Info_Fields, allocator: runtime.Allocator) -> (info: Process_Info, err: Error) {
-	return
+	return _process_info_by_pid(process.pid, selection, allocator)
 }
 
 @(private="package")
 _current_process_info :: proc(selection: Process_Info_Fields, allocator: runtime.Allocator) -> (info: Process_Info, err: Error) {
-	return
+	return _process_info_by_pid(get_pid(), selection, allocator)
 }
 
 @(private="package")
-_process_open :: proc(pid: int, flags: Process_Open_Flags) -> (process: Process, err: Error) {
+_process_open :: proc(pid: int, _: Process_Open_Flags) -> (process: Process, err: Error) {
+	process.pid = pid
+	process.handle = PIDFD_UNASSIGNED
+	if !_has_pidfd_open {
+		return process, .Unsupported
+	}
+
+	pidfd, errno := linux.pidfd_open(linux.Pid(pid), {})
+	if errno == .ENOSYS {
+		_has_pidfd_open = false
```

Comment:
Should this be atomic or `@(thread_local)`? Because it's a global variable that could be accessed from multiple threads.